File: app/src/main/java/com/github/pokatomnik/frontendconf2023/examples/ex01functions/ex01Function.ts
export class SomeClass {
  public constructor(public readonly value: string) {}
}

export function example(
  a: number,
  b: string | null,
  c: boolean = false,
  foo: SomeClass | null,
  callback: (result: string) => void
) {
  const result = `${a}, ${b}, ${c},`;
  callback(`${result} ${foo?.value}`);
}

File: app/src/main/java/com/github/pokatomnik/frontendconf2023/examples/ex01functions/ex01Function.kt
package com.github.pokatomnik.frontendconf2023.examples.ex01functions

data class SomeClass(val value: String)

fun example(
    a: Int,
    b: String?,
    c: Boolean = false,
    foo: SomeClass?,
    callback: (result: String) -> Unit,
) {
    val result = "$a, $b, $c,"
    callback("$result ${foo?.value}")
}